Source/Core/VideoBackends/Vulkan/VKSwapChain.cpp

```diff
@@ -176,18 +176,24 @@ bool SwapChain::SelectSurfaceFormat()
     // because we already apply gamma ourselves, and we might not use sRGB gamma.
     // Force using a linear format instead, if this is the case.
     VkFormat format = VKTexture::GetLinearFormat(surface_format.format);
-    if (format == VK_FORMAT_R8G8B8A8_UNORM)
-      surface_format_RGBA8 = &surface_format;
-    else if (format == VK_FORMAT_B8G8R8A8_UNORM)
-      surface_format_BGRA8 = &surface_format;
-    else if (format == VK_FORMAT_A2B10G10R10_UNORM_PACK32 &&
-             surface_format.colorSpace == VK_COLOR_SPACE_SRGB_NONLINEAR_KHR)
-      surface_format_RGB10_A2 = &surface_format;
+    if (surface_format.colorSpace == VK_COLOR_SPACE_SRGB_NONLINEAR_KHR)
+    {
+      if (format == VK_FORMAT_R8G8B8A8_UNORM)
+        surface_format_RGBA8 = &surface_format;
+      else if (format == VK_FORMAT_B8G8R8A8_UNORM)
+        surface_format_BGRA8 = &surface_format;
+      else if (format == VK_FORMAT_A2B10G10R10_UNORM_PACK32)
```

Comment:
This needs a:

```
else
  continue;
```

Reply:
Actually, didn't look at the context.  No it should be fine to fall through.

Though makes me wonder why we have a `else continue` in the other branch :/

Reply:
Ah, yes. I've removed the unnecessary continue.